### PR TITLE
Delete challenges

### PR DESCRIPTION
**Pull request by mwinstead3790**
I have implemented Enhancement issue #15 . If acceptable, I can remove the debugging lines (which are there so you can see what is going on) or change anything as needed.

### CODE REVIEW

File: CTFd/admin.py [hunk 1]
```diff
@@ -477,6 +480,47 @@ def admin_create_chal():
         db.session.close()
         return redirect('/admin/chals')
 
+    @app.route('/admin/chal/delete', methods=['POST'])
+    def admin_delete_chal():
+        challenge=Challenges.query.filter_by(id=request.form['id']).first()
```
Comment (ColdHeat) [inline CTFd/admin.py]: Needs spacing around the assignment equals sign.

File: CTFd/admin.py [hunk 2]
```diff
@@ -477,6 +480,47 @@ def admin_create_chal():
...
+        if challenge is not None:
```
Comment (ColdHeat) [inline CTFd/admin.py]: Just needs to be `if challenge:`

File: CTFd/admin.py [hunk 3]
```diff
@@ -477,6 +480,47 @@ def admin_create_chal():
...
+            #print challenge, " id: ", challenge.id
+            #Clean up wrong keys
+            #wrong_keys=WrongKeys.query.filter(chal=challenge.id).all()
+            wrong_keys= db.session.query(WrongKeys, Challenges).filter(Challenges.id==WrongKeys.id, Challenges.id==challenge.id).all()
```
Comment (ColdHeat) [inline CTFd/admin.py]: Pretty sure this should be `challenge.id==WrongKeys.chal`

**Comment by ColdHeat** [comment]
Thanks so much for the PR! I left you some comments in the files changed. But I have a feeling more of those queries are incorrect and I think with a portion of CTFd this sensitive I'd rather implement it on my own. 

I'll be sure to credit you in my commit which should be in a few hours.

**Comment by mwinstead3790** [comment]
Hey, sorry for the first pep8 violation, but the if statement is pep8 programming recommendation 2 [1].

Also, the database operation is selecting from the Challenge.id foreign keys. I tried for a couple of hours just fetching with something like the following:
```python
wrong_keys = db.session.query(WrongKeys).filter(WrongKeys.id==challenge.id).all()
```

but I couldn't get rows returned. Thus, I read more about foreign keys on the object relational tutorial for SQLAlchemy 0.9 and when I tried the joins [2], all was good. Thus, this is just making a join table between the two tables and then setting a condition on the Challenges.id field:

```python
wrong_keys= db.session.query(WrongKeys, Challenges).filter(Challenges.id==WrongKeys.id, Challenges.id==challenge.id).all()
 ```
[1] http://docs.sqlalchemy.org/en/rel_0_9/orm/tutorial.html#querying-with-joins
[2]https://www.python.org/dev/peps/pep-0008/#programming-recommendations

Regardless - thanks for responding to the pull request!
-Mike

**Comment by mwinstead3790** [comment]
Gotcha - thanks for getting to looking at the pull request.

I'll probably also implement user promotion and page deletion as well. I'll go back and test/fix these in my fork if you ever want to check it back out - I see where I missed a couple of things. 

Thanks again,
-Mike

**Pull request closed without merging**